Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -319,7 +319,8 @@ void InsertPageBefore(Page page, Page before)
 
 			var root = (Page)PageController.InternalChildren.First();
 
-			PageController.InternalChildren.ToArray().Where(c => c != root).ForEach(c => PageController.InternalChildren.Remove(c));
+            var childrenToRemove = PageController.InternalChildren.ToArray().Where(c => c != root);  
+            childrenToRemove.ForEach(c => PageController.InternalChildren.Remove(c)); 
```

Comment:
No problem, I just figured it was the preferred way since that's the way it looked in the original forms code. I'll fix the changes and squash it at the end after the code passes you review.